packages/grafana-ui/src/components/Table/TableNG/Cells/AutoCell.tsx

```diff
@@ -1,10 +1,62 @@
-import { formattedValueToString } from '@grafana/data';
+import { css } from '@emotion/css';
+import { useState, useRef } from 'react';
 
+import { GrafanaTheme2, formattedValueToString } from '@grafana/data';
+
+import { useStyles2, useTheme2 } from '../../../../themes';
 import { CellNGProps } from '../types';
 
-export default function AutoCell({ value, field, height, justifyContent }: CellNGProps) {
+export default function AutoCell({ value, field, justifyContent, shouldTextOverflow }: CellNGProps) {
   const displayValue = field.display!(value);
   const formattedValue = formattedValueToString(displayValue);
 
-  return <div style={{ display: 'flex', justifyContent }}>{formattedValue}</div>;
+  // const theme = useTheme2();
+  const styles = useStyles2(getStyles);
+
+  const divRef = useRef<HTMLDivElement>(null);
+
+  const [isHovered, setIsHovered] = useState(false);
+
+  const handleMouseEnter = () => {
+    if (shouldTextOverflow && !shouldTextOverflow()) {
+      return;
+    }
+    setIsHovered(true);
+
+    const div = divRef.current;
+    div?.parentElement?.style.setProperty('overflow', 'visible');
+  };
+
+  const handleMouseLeave = () => {
+    if (shouldTextOverflow && !shouldTextOverflow()) {
+      return;
+    }
+
+    setIsHovered(false);
+
+    const div = divRef.current;
+    div?.parentElement?.style.setProperty('overflow', 'hidden');
+  };
+
+  return (
+    <div
+      ref={divRef}
+      style={{ display: 'flex', justifyContent }}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+      className={isHovered ? styles.cell : ''}
+    >
+      {formattedValue}
+    </div>
+  );
 }
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  cell: css({
+    whiteSpace: 'break-spaces',
+    background: theme.colors.background.primary,
+    border: `1px solid ${theme.colors.border.medium}`,
+    zIndex: 999,
```

Comment:
Where does this number come from? I noticed cell inspect still works and did a little digging. Z-index is defined in this file. Let's use it!

https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/themes/zIndex.ts

Reply:
Oh, cool, thank you for finding those constants 🙌🏻